biojava-structure/src/test/java/org/biojava/nbio/structure/asa/TestAsaCalc.java

```diff
@@ -70,12 +70,114 @@ public void testAsa3PIU() throws StructureException, IOException {
 			//System.out.println(groupAsa.getGroup().getPDBName() + " " + groupAsa.getGroup().getResidueNumber() + " " + groupAsa.getAsaU());
 			totResidues+=groupAsa.getAsaU();
 
-			Assert.assertTrue(groupAsa.getRelativeAsaU() <= 1.0);
+			assertTrue(groupAsa.getRelativeAsaU() <= 1.0);
+		}
+
+		assertEquals(totAtoms, totResidues, 0.000001);
+
+		assertEquals(17462.0, totAtoms, 1.0);
+
+	}
+
+	@Test
+	public void testNeighborIndicesFinding() throws StructureException, IOException {
+		// important: without this the tests can fail when running in maven (but not in IDE)
+		// that's because it depends on the order on how tests were run - JD 2018-03-10
+		ChemCompGroupFactory.setChemCompProvider(new DownloadChemCompProvider());
+
+		Structure structure = StructureIO.getStructure("3PIU");
+
+		AsaCalculator asaCalc = new AsaCalculator(structure,
+				AsaCalculator.DEFAULT_PROBE_SIZE,
+				1000, 1, false);
+
+		int[][] allNbsSh = asaCalc.findNeighborIndicesSpatialHashing();
+
+		int[][] allNbs = asaCalc.findNeighborIndices();
+
+		for (int indexToTest =0; indexToTest < asaCalc.getAtomCoords().length; indexToTest++) {
+			//int indexToTest = 198;
+			int[] nbsSh = allNbsSh[indexToTest];
+			int[] nbs = allNbs[indexToTest];
+
+			List<Integer> listOfMatchingIndices = new ArrayList<>();
+			for (int i = 0; i < nbsSh.length; i++) {
+				for (int j = 0; j < nbs.length; j++) {
+					if (nbs[j] == nbsSh[i]) {
+						listOfMatchingIndices.add(j);
+						break;
+					}
+				}
+			}
+			
+//		for (int i = 0; i<nbs.length; i++) {
+//			double dist = asaCalc.getAtomCoords()[i].distance(asaCalc.getAtomCoords()[indexToTest]);
+//			if (listOfMatchingIndices.contains(i)) {
+//				System.out.printf("Matching     - indices %d-%d: %5.2f\n", indexToTest, i, dist);
+//			} else {
+//				System.out.printf("Not matching - indices %d-%d: %5.2f\n", indexToTest, i, dist);
+//			}
+//		}
+
+			assertEquals(nbs.length, nbsSh.length);
+
+			assertEquals(nbs.length, listOfMatchingIndices.size());
```

Comment:
I'd probably have two tests.
1. test for the functionality against the simplest implementation. 
2. test ensuring the two implementations produce equivalent results.
I am only reading the diffs, so I may be overlooking what is already there.